### PR TITLE
Suppress debug logging and use block-local stream

### PR DESCRIPTION
**Pull request by stuarthicks**

### CODE REVIEW

File: libsmile.mk [hunk 1]
```diff
@@ -36,7 +36,7 @@ LIBTOOL  = libtool
 endif
 
 SMILE_INCLUDE = -I$(CURDIR)/include
-CFLAGS   += -std=c99 -pedantic -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer -I. $(SMILE_INCLUDE)
+CFLAGS   += -std=c99 -pedantic -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer -I. $(SMILE_INCLUDE) -g
```
Comment (stuarthicks) [inline libsmile.mk:39]: This is optional, it puts debug symbols into both `libsmile.a` and `unsmile`... which let me gdb it more easily when it was segfaulting.

File: lib/error.h [hunk 2]
```diff
@@ -32,7 +33,6 @@
 #define ERROR_REPORT(prefix, msg, error) \
     do { \
         /* TODO Should go in strm->msg */ \
-        fprintf(stderr, "%s: %s\n", prefix, msg); \
```
Comment (stuarthicks) [inline lib/error.h:35]: This removes some pretty useless debug logging.

File: tools/unsmile.c [hunk 3]
```diff
@@ -27,7 +27,7 @@
 #include "usage.h"
 #include <smile.h>
 
-#define BUFFER_SIZE 65536
+#define BUFFER_SIZE 32768
```
Comment (stuarthicks) [inline tools/unsmile.c:30]: I'm confident we don't need buffers quite so large, given the size of what we're parsing.

File: lib/error.h [hunk 4]
```diff
@@ -42,4 +42,4 @@
 #define RESERVED(s)         ERROR_REPORT("RESERVED", s, -EILSEQ)
 #define NOT_IMPLEMENTED(s)  ERROR_REPORT("NOT IMPLEMENTED", s, -ENOSYS)
 
-#endif
\ No newline at end of file
+#endif
```
Comment (thomshutt) [inline lib/error.h:45]: 👌

**Review by thomshutt — APPROVED** [review]

**Pull request merged**